plugins/by-name/ltex-extra/default.nix

```diff
@@ -43,26 +43,60 @@ lib.nixvim.plugins.mkNeovimPlugin {
     ] "Log level.";
   };
 
-  extraConfig = cfg: {
-    warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" {
-      when = !config.plugins.lsp.enable;
-      message = ''
-        You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+  extraConfig =
+    cfg:
+    let
+      inherit (config.plugins.lsp.servers) ltex;
+      attachLua = ''
+        require("ltex_extra").setup(${lib.nixvim.toLuaObject cfg.settings})
       '';
-    };
-
-    plugins.lsp = {
-      servers.ltex = {
-        # Enable the ltex language server
-        enable = true;
+    in
+    {
+      warnings = lib.nixvim.mkWarnings "plugins.ltex-extra" [
+        {
+          when = !config.plugins.lsp.enable;
+          message = ''
+            You have enabled `ltex-extra` but not the lsp (`plugins.lsp`).
+            You should set `plugins.lsp.enable = true` to make use of the LTeX_extra plugin's features.
+          '';
+        }
+        {
+          # TODO: Added 2025-03-30; remove after 25.05
+          # Warn if servers.ltex seems to be configured outside of ltex-extra
+          when =
+            !ltex.enable
+            && (
+              ltex.onAttach.function != attachLua
+              || ltex.onAttach.override != false
```

Comment:
I wonder if it's better to check these options' override priority, to assert that our default value isn't being overridden rather than explicitly checking the actual value.

E.g. `ltexOpt.highestPrio < (mkDefault null).priority` if we were comparing against a `mkDefault` definition.

Reply:
It can probably be done that way. But it is no longer necessary as I rewrote this to check for definitions in any unexpected files instead.